### PR TITLE
Create job token by default

### LINKED ISSUE #44: GirderClientTransforms for expired sessions.
There should be a way to deal with upload transforms that happen after long-running jobs, when it's possible that the user that started the job has ended their web session.

Easiest thing I can think of is to allow passing alternative credentials, like an API key, to the transform constructor.

Also, this behavior isn't documented and it should be.  I would actually expect API token generation to be the default behavior.  I'm inclined to call this a bug.

### PR DESCRIPTION
**Pull request by subdavis**
closes #44

I looked at adding a test, but `gc=None` is not a condition that was ever tested before, so if we want to add that here I can look into it more.

Tested locally working.

Edit: notice in particular the scope restrictions.  I think these are safe, but I'm not positive.

### CODE REVIEW

File: girder_worker_utils/transforms/girder_io.py [hunk 1]
```diff
@@ -28,9 +28,21 @@ def __init__(self, *args, **kwargs):
...
+
                 self.gc = GirderClient(apiUrl=getWorkerApiUrl())
-                self.gc.token = getCurrentToken()['_id']
+                from girder.api.rest import getCurrentUser
+                if getCurrentUser():
+                    from girder.constants import TokenScope
+                    from girder.models.token import Token
+                    token = Token().createToken(
+                        days=1,
+                        scope=[TokenScope.DATA_READ, TokenScope.DATA_WRITE],
```
Comment (zachmullen) [inline girder_worker_utils/transforms/girder_io.py]: My concern with using this scope is that it might not allow writes back to the job itself. I could be wrong about that though... Have you tried it?

Comment (subdavis) [reply girder_worker_utils/transforms/girder_io.py]: Yes, I've tested this and it writes the status back to the job just fine.

**Comment by subdavis** [comment]
Updated to 7 days expiration.

File: girder_worker_utils/transforms/girder_io.py (hunk 1 above)
Comment (zachmullen) [reply girder_worker_utils/transforms/girder_io.py]: Great. I sanity checked the relevant code and that makes sense since this is a user-associated token:

```python
    def updateJob(self, job, log, overwrite, notify, status, progressTotal, progressCurrent,
                  progressMessage):
        user = self.getCurrentUser()
        if user:
            self._model.requireAccess(job, user, level=AccessType.WRITE)
        else:
            self.ensureTokenScopes('jobs.job_' + str(job['_id']))
```

**Comment by subdavis** [comment]
I'm not authorized to merge.
